src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java

```diff
@@ -51,14 +55,32 @@ public void save(
 
         caseRepository.saveClaimantResponse(claim, response, authorization);
 
-        if (ClaimantResponseType.ACCEPTATION.equals(response.getType())) {
-            formaliseResponseAcceptanceService.formalise(claim, (ResponseAcceptation) response, authorization);
-        }
+        formaliseResponseAcceptance(response, claim, authorization);
 
         eventProducer.createClaimantResponseEvent(claim);
         appInsights.trackEvent(getAppInsightsEvent(response), claim.getReferenceNumber());
     }
 
+    private void formaliseResponseAcceptance(ClaimantResponse claimantResponse, Claim claim, String authorization) {
+        Response response = claim.getResponse().orElseThrow(IllegalStateException::new);
+
+        if (ClaimantResponseType.ACCEPTATION.equals(claimantResponse.getType())
+            && !isResponseStatesPaid(response)) {
+            formaliseResponseAcceptanceService.formalise(claim, (ResponseAcceptation) claimantResponse, authorization);
+        }
+    }
+
+    private boolean isResponseStatesPaid(Response response) {
+        switch (response.getResponseType()) {
```

Comment:
Suggestion :- Use switch if there are more no of cases. For just two types, use if else.